lib/request.js

```diff
@@ -313,8 +313,13 @@ AWS.Request = inherit({
     var region = service.config.region;
     var customUserAgent = service.config.customUserAgent;
 
-    // global endpoints sign as us-east-1
-    if (service.isGlobalEndpoint) region = 'us-east-1';
+    if (service.isGlobalEndpoint) {
+      if (service.signatureRegion) {
```

Comment:
`signingRegion` can be applicable for non-global regions right?

Reply:
If the service is non-global the `signingRegion` should be equal to the region passed to the client. Either the detected region or the explicitly specified one.